File: QE-Selenium-Java/Activity16.java
package demos;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.Select;

import java.util.List;

public class Activity16 {
    public static void main(String[] args) {

        WebDriver driver = new FirefoxDriver();
        driver.get("https://training-support.net/webelements/selects");

        System.out.println(driver.getTitle());

        WebElement singleSelectElement = driver.findElement(By.tagName("select"));
        Select singleSelect = new Select(singleSelectElement);

        singleSelect.selectByVisibleText("Two");
        singleSelect.selectByIndex(2);
        singleSelect.selectByValue("four");

        System.out.println("All options in Single Select:");
        List<WebElement> options = singleSelect.getOptions();
        for (WebElement option : options) {
            System.out.println(option.getText());
        }

        driver.quit();
    }
}

File: QE-Selenium-Python/Activity16.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import Select

driver = webdriver.Firefox()
driver.get("https://training-support.net/webelements/selects")
print(driver.title)
single_select_element = driver.find_element(By.TAG_NAME, "select")
single_select = Select(single_select_element)
single_select.select_by_visible_text("Two")
single_select.select_by_index(2)
single_select.select_by_value("four")
print("All options in Single Select:")
options = single_select.options
for option in options:
    print(option.text)
driver.quit()